handlers/messageHandlers.js

```diff
@@ -25,6 +26,10 @@ const commandParser = async (client, con, msg) => {
     return msg.reply(`I can't execute that command inside DMs!`);
   }
 
+  if (command.staffOnly && !hasPermission(msg, command)) {
```

Comment:
Just to clarify (my JS knowledge is not as thorough as I'd like it to be), if the `command` object does not have a `staffOnly` attribute, this if statement should still execute correctly, right?

To clarify my current understanding, if a basic user in the server tried to run a moderator command (without the `staffOnly` attribute), the if statement would effectively be `if (null && true)`. Is that understanding accurate, or would we see an error for the missing attribute rather than a null value?

Reply:
Unfortunately, because this is JS, the the `staffOnly` attribute does not exist, it returns `undefined` and never enters conditional block.  For this reason, we'd need to make sure each staff command had the `staffOnly` attribute for all future commands.

Because the current staff commands haven't yet been converted to use this permissions handler, this `if` block will return `undefined` and then the command will continue trying to execute, only to run into something like this inside the actual command:
https://github.com/CodecademyCommunity/codecademy-discord-bot/blob/5440cc0283e65ad5298448b27a7425fd762f6616/commands/mute/mute.js#L32-L40

Once the staff commands are converted (issues #123, #124, #125) they should all contain both `staffOnly` and `minRole` attributes.

BUT, you bring up a fantastic point, and maybe it can help me sway the team to port this over TypeScript in the future 😄

Reply:
Thanks for the detailed explanation!